### PR TITLE
feat: consistent index result output

### PR DESCRIPTION
**Pull request by V0ldek**
The `--result bytes` mode now consistently reports the first byte of the value it matched. This can be used to extract the actual value from the JSON by parsing from the reported byte.

## Issue

Resolves: #161

## Checklist

All of these should be ticked off before you submit the PR.

- [x] I ran `just verify` locally and it succeeded.
- [x] Issue was given <span style="color: #FF4400">go ahead</span> and is linked above **OR** I have included justification for a minor change.
- [x] Unit tests for my changes are included **OR** no functionality was changed.

### CODE REVIEW

**Review by zwerdlds — COMMENTED** [review]
Just a couple minor things to take a look at.

File: crates/rsonpath-lib/src/result.rs [hunk 1]
```diff
@@ -1,24 +1,101 @@
...
+    /// {
+    ///   "match": [42,     30]
+    ///   //          ^^^^^^^
+    ///   // any of these characters can be reported for the query $.match[1]
+    /// }
+    /// ```
+    ///
+    /// # Errors
+    /// This function may access the input, which can raise an [`EngineError::InputError`].
+    /// Mo
```
[the remark this answers is not in the document]
Comment (V0ldek) [reply crates/rsonpath-lib/src/result.rs]: Actually just forgot to write the rest of the doc, lol. Fixed now.

File: crates/rsonpath-lib/src/input.rs [hunk 2]
```diff
@@ -183,6 +201,46 @@ pub(super) mod in_slice {
...
+            }
+            idx += 1;
+            if idx == bytes.len() {
+                return None;
+            }
+        }
+    }
+
+    #[inline]
+    pub(super) fn seek_forward_2(bytes: &[u8], from: usize, needle_1: u8, needle_2: u8) -> Option<usize> {
```
[the remark this answers is not in the document]
Comment (V0ldek) [reply crates/rsonpath-lib/src/input.rs]: This is much better, thank you.

Also apparently it results in better codegen?? Compare the old `seek_forward`:

```asm
        push rax
        cmp rdx, rsi
        jae .LBB220_1
        mov r8, rdx
        cmp rsi, r8
        je .LBB220_7
        mov rdx, r8
        cmp byte ptr [rdi + r8], cl
        je .LBB220_6
        lea r8, [rdx + 1]
        cmp rsi, r8
        pop rcx
        ret

.LBB220_1:
        xor eax, eax
        pop rcx
        ret

.LBB220_6:
        mov eax, 1
        pop rcx
        ret

.LBB220_7:
        .cfi_def_cfa_offset 16
        lea rdx, [rip + .Lanon.13c2e7e34fc5d699ce710eb22118683a.72]
        mov rdi, rsi

        call qword ptr [rip + core::panicking::panic_bounds_check@GOTPCREL]

        ud2
```

with the new `seek_forward::<1>`:

```asm
        mov rax, rdi
        xor edi, edi
        cmp rcx, rdx
        jae .LBB220_4
.LBB220_1:
        cmp byte ptr [rsi + rcx], r8b
        je .LBB220_3
        inc rcx
        cmp rdx, rcx
        jne .LBB220_1

.LBB220_4:
        mov qword ptr [rax], rdi
        ret

.LBB220_3:
        mov edi, 1
        mov qword ptr [rax + 8], rcx
        mov byte ptr [rax + 16], r8b
        mov qword ptr [rax], rdi
        ret
```

Doesn't use the stack, eliminated the bounds-check all together... Compilers are such weird animals.

[the remark this answers is not in the document]
Comment (V0ldek) [reply crates/rsonpath-lib/src/input.rs]: This is much better, thank you.

Also apparently it results in better codegen?? Compare the old `seek_forward`:

```asm
        push rax
        cmp rdx, rsi
        jae .LBB220_1
        mov r8, rdx
        xor eax, eax
.LBB220_4:
        cmp rsi, r8
        je .LBB220_7
        mov rdx, r8
        cmp byte ptr [rdi + r8], cl
        je .LBB220_6
        lea r8, [rdx + 1]
        cmp rsi, r8
        jne .LBB220_4
        pop rcx
        ret

.LBB220_1:
        xor eax, eax
        pop rcx

        ret

.LBB220_6:
        mov eax, 1
        pop rcx

        ret

.LBB220_7:
        lea rdx, [rip + .Lanon.6db0d29e2c8f457d477730b4ec9a7c5c.72]
        mov rdi, rsi

        call qword ptr [rip + core::panicking::panic_bounds_check@GOTPCREL]

        ud2
```

with the new `seek_forward::<1>`:

```asm
        mov rax, rdi
        xor edi, edi
        cmp rcx, rdx
        jae .LBB220_4
.LBB220_1:
        cmp byte ptr [rsi + rcx], r8b
        je .LBB220_3
        inc rcx
        cmp rdx, rcx
        jne .LBB220_1

.LBB220_4:
        mov qword ptr [rax], rdi
        ret

.LBB220_3:
        mov edi, 1
        mov qword ptr [rax + 8], rcx
        mov byte ptr [rax + 16], r8b
        mov qword ptr [rax], rdi
        ret
```

Doesn't use the stack, eliminated the bounds-check all together... Compilers are such weird animals.

**Review by zwerdlds — COMMENTED** [review]
LGTM

**Review by zwerdlds — APPROVED** [review]

**Pull request merged**